### ISSUE TITLE
calcNhoodExpression doesn't work for a single feature

### ISSUE THREAD

**Issue by emdann** [MikeDMorgan/miloR#32]
Maybe it's enough to throw an informative error message, but at the moment running with one gene gives a funny error.

```r
> calcNhoodExpression(sim1.mylo, subset.row = c("Gene1", "Gene2"))
class: Milo 
dim: 1000 1100 
metadata(0):
assays(1): logcounts
rownames(1000): Gene1 Gene2 ... Gene999 Gene1000
rowData names(0):
colnames(1100): Cell1 Cell2 ... Cell1099 Cell1100
colData names(0):
reducedDimNames(1): PCA
spikeNames(0):
altExpNames(0):
nhoods dimensions(1): 100
nhoodCounts dimensions(2): 100 6
nhoodDistances dimensions(2): 1100 1100
graph names(1): graph
nhoodIndex names(1): 100
nhoodExpression dimension(2): 2 100
nhoodReducedDim names(0):
nhoodGraph names(0):
```

```
> calcNhoodExpression(sim1.mylo, subset.row = c("Gene1"))
Error in t(neighbour.model) %*% t(data.set[subset.row, ]) : 
  non-conformable arguments
```

It's probably just about adding a `drop=FALSE`. @MikeDMorgan I can fix this in `.calc_expression` if that's ok with you.

**Comment by MikeDMorgan** [comment]
Yeah, it's because R will automatically cast a 1-D matrix to a vector so the dimensions won't match. The alternative approach would be to check the class of the subset object first, then perform the correct operation based on that result.

**Comment by emdann** [comment]
Changing all instances of `data.set[subset.row, ]` in `.calc_expression` to `data.set[subset.row, , drop=FALSE]` seems to do the job. Is there any side effect I am not seeing?

**Comment by MikeDMorgan** [comment]
That's definitely the simplest way to do it.

**Comment by emdann** [comment]
Fixed with 064ebd4
